### DISCUSSION TITLE
Hyperion Lighting Build 2023:

### DISCUSSION THREAD

**Discussion by 53times** [awawa-dev/HyperHDR#577] [Q&A]
It seems to me there’s a lot of people wanting ambient lighting on their TV’s and I refuse to think I’m not smart enough to follow instructions. Have any of you considered possibly doing a “How To” series on the idea? Maybe charge for your services?

My goal is to have ambient lighting behind my TV. I have an 85” Samsung and thought I’d simply purchase some sk6812 lights and a Raspberry Pi 3B+. Learn something new and do something in retirement. So I bought the Pi and 2 sets of BTF sk6812 5m lights. The TV will use about 6m. These are 5v and 60 leds per meter.

After a couple of weeks of hit and miss… I finally got HyperHDR on my Pi. I can SSH in and get the HyperHDR set-up page. (I’m explaining this in detail because I know nothing at all about what I’m doing… I’ve spent too much time and money and I’m getting frustrated with no positive results.) In the HyperHDR set-up I noticed there was no “controller” type for my sk6812 lights, only for the sk6812spi products. So I selected it and received a message window stating I shouldn’t use this “controller”. Then I learned I also needed an ESP32 board. Then came brands and specifics I know nothing about. WLED, HyperSPI, CH9102X, MS2130,etc. I’d like something that’s good quality and built to last. I’ve read a lot of posts here and watched a lot of youtube. I learned the videos are outdated or assume I know more than I do. The instructions on Github are good but confusing to me because I don’t know what micro computer to get or programs to download/flash. Seems every time I try something I read it doesn’t work. 

Ultimately when this idea comes to fruition, I’d like to tackle bigger projects but believe this one will get my feet wet enough to know if I want to continue. Any volunteers to help me through this? Probably a lot of stupid questions on my part, but your time would be appreciated.

If you’re willing… please ask questions. 

Why would I want a WLED over a non WLED connection? Is one any better than the other? Faster? Better processor? Better quality? If you were building a system from scratch now… what specific parts would you use considering what I now have? Any hardware recommendations would be great. Seems to be all kinds of software to look for… any guidance? Is there any way to control on/off for a soft shutdown of the Pi and associated system from my phone? Simple is good.

Kindness is appreciated. I appreciate your valuable time. If you feel this post is too long, irrelevant, or inappropriate… please delete it. OR steer me towards someone who might be available to take the time to help. Thanks in advance.

**Comment by awawa-dev** [comment]
I'm moving it to the discussion section. You have the latest tutorial at https://www.hyperhdr.eu/2023/02/ultimate-guide-on-how-to-build-led.html Is there anything unclear about it? It's 3 months old but of course there has been steady progress recently as the ms2130 seems to be working really well. There is also a very promising rp2040 that could potentially replace the esp board solution due to its very good signal quality and very easy flashing procedure, especially as some boards have built-in 3.3v to 5v level shifter: you can read more about it here https://github.com/awawa-dev/HyperHDR/discussions/561

Wi-Fi LED devices are generally not recommended for use with HyperHDR for constant TV ambient lighting streaming, only wired solution. Usually it's hard to maintain constant low-latency wifi connection over long time for esp boards. And in no case should you run the sk6812 directly from the Rpi as stated in the FAQ: you need esp/rp2040 board + 3.3v to 5v level shifter.

More user-made tutorials are of course very welcome.

**Reply by 53times** [reply]
Thank you Mam/Sir. Yes I’ve read the tutorial a couple of times and learned some. But it also got me thinking that the rp2040 is the way to go. Soo I have a question… it seems the only way to use the 2040 currently is if you flash the unit with “HyperHDR v20beta0”. Is it that simple? I’m not a fan of Beta versions and would rather just do the install once. I had to wipe the SD numerous times just to get where I am now with HyperHDR dashboard. (Too much ambiguity and out of date info) So would I be better off just waiting for the tested version? Maybe installing “HyperSerialESP32” instead? And switching to proven v20 at a later date? Or is there a lot of info that will need to be transferred from one program to the other? I believe the v20 version will readily address my needs. Because of the distance I’m covering with the sk6812’s... I’m planning on injecting power at both ends of two equal strips. I’m thinking that would equate to a “multi-segment strip” which is what the tutorial referred to.  Another of my questions was how to address the data conductor? It seems the rp2040 would be able to control the data whether I series or parallel it? Whichever provides better efficiency and synchronization in your opinion? Hope that makes sense.

On a side note… I appreciate your time. I’m a believer in bartering. I’m a retired Master Electrician and have quite a bit of experience in electrical design and installation. (but I don’t know about computers! Data centers and linear power supplies… yes.) If you have an electrical question… I’d be happy to give you an answer. Immediately. If I don’t know the answer off the top of my head I’ll research it till I know it fits your needs. And if I just flat don’t know the answer… I won’t baffle you. I’ll tell you straight out. Of course my answers would be recommendations and NOT specific advice due to legalities. 

Thank you for your time and knowledge. P.S. I do have a level shifter… but if the 2040 has one… mine will be a spare.

**Reply by awawa-dev** [reply]
Hi

Yes, you must be using the latest beta version of HyperHDR v20. The stable version is planned in a few months at the earliest (now planning a summer break), but I use v20 on my home system almost every day. If there are bugs that prevent the beta from working properly, I try to fix them immediately. It's always a good idea to back up your configuration (HyperHDR General tab -> export configuration as json file): in case of problems with v20 beta, you can flash the SD v19 image again and restore it that way. Or preparing a second SD card is an even simpler solution.

rp2040 allows multi-segment control: the first segment is on the selected gpio: OUTPUT_DATA_PIN , the second segment is on OUTPUT_DATA_PIN + 1. If you have a level shifter, it's better to use it with the rp2040 or esp32 instead of buying dedicated rp2040 boards with a built-in level shifter, because these boards are more expensive than generic solution and you can chose between esp32 or rp2040. They are rather a great and simple solution for users who, for various reasons, cannot add a level shifter to their LED controller on their own. I will gladly use the consultation in the future if I have any doubts about new projects :)

**Comment by AstaRom** [comment]
> Why would I want a WLED over a non WLED connection? Is one any better than the other? Faster? Better processor? Better quality?

HyperSerialWLED - wireless connection, there is a slight delay.

HyperSerialEsp8266 / HyperSerialESP32 / HyperSerialPico - Usb wire connection, lower latency compared to WLED. It is recommended to use when there is a long distance from the Raspberry Pi to the microcontroller.

HyperSPI - connection using Dupont jumpers. Least delay. Recommended if the distance from the Raspberry Pi to the microcontroller is 15-20 cm.


> OR steer me towards someone who might be available to take the time to help.

I don't know how much this will help you as Awawa answered above, there is a guide on his blog. Recently, I also began to collect information about my experience in assembling the backlight [in a small article](https://astarom.github.io/index_en.html) . Of course, it is not as detailed and not completely complete, but maybe this information will be of some use to you))

**Reply by 53times** [reply]
AstraRom… Thank you! Your article is really good and informative. It helped quite a bit. I had heard/read so much about the WLED I thought I was missing something when it came to performance. I appreciate your validating the time delay. My Pi 3B+ and Pico will be within the 20cm limitation. I’ll be checking your “article” from time to time for updates? I’d like to know more about your experiences with your build. Are you a newbie or relatively experienced? As far as I’m concerned… successfully building and operating one ambient light project makes you experienced!

Please read my response to awa-dev above… I’ll make you the same deal if you have a question.

**Reply by AstaRom** [reply]
> Please read my response to awa-dev above… I’ll make you the same deal if you have a question.

I myself work in the same field as you ))
<details>
   <summary></summary>
  
![20230527_053622](https://github.com/awawa-dev/HyperHDR/assets/39379875/a3eeab08-dbbe-45f4-ac63-e118b63a196c)
![20230527_061645](https://github.com/awawa-dev/HyperHDR/assets/39379875/06adda76-f72c-4cd9-aae5-6c861819d4d0)

</details>

> Are you a newbie or relatively experienced?

I had previous experience with various controllers which was useful in this project as well. Well, I also had the skills to work with a soldering iron and printed circuit boards before that.

**rp2040** - this is a promising device, as respected Awawa already mentioned. I probably if I were assembling a new assembly considering it for buy. But it is worth considering that the driver (HyperSerialPico) for it can be said to be a Beta-version.
